internal/ethapi/api.go

```diff
@@ -2044,6 +2044,24 @@ func runMEVM(ctx context.Context, b Backend, state *state.StateDB, header *types
 		computeResult = result.ReturnData // Or should it be nil maybe in this case?
 	}
 
+	logs := state.GetLogs(common.Hash{}, 0, common.Hash{})
+
+	if len(logs) != 0 {
+		result := suave.ExecResult{
+			Logs: logs,
+		}
+
+		logsEncoded, err := result.EncodeABI()
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		// add some magic number and the result (TODO)
+		computeResult = append(computeResult, []byte{0xff}...)
+		computeResult = append(computeResult, logsEncoded...)
```

Comment:
I guess an interesting alternative would be to make the log coalescing explicit/configurable
As an example we could add a precompile that would return the coalesced logs, or have the logs live in the context and add a simple solidity wrapper for encoding the logs to the result of a function
This way we don't have to change the client code to change how logs are added and parsed

Reply:
Something like?

```
function onchain(uint256 num) {
}

function offchain() {
   emit Event();

   return abi.encodeWithSelector(this.onchain.selector, 1, Suave.getLogs());
}
```

Well, you still need to create a new precompile and there is the work of teaching users to put that last argument. But I think making it implicit also leans more towards the side effect model. Also, the changes are not that big on the client.